CMake/SlicerDashboardDriverScript.cmake

```diff
@@ -566,26 +566,29 @@ ${ADDITIONAL_CMAKECACHE_OPTION}
             BUILD ${slicer_build_dir}
             APPEND
             )
-          ctest_submit(PARTS Build)
+          if(run_ctest_submit)
+            ctest_submit(PARTS Build)
+          endif()
         endif()
 
         if(run_ctest_with_upload)
           message(STATUS "Uploading Slicer package URL ...")
 
           file(STRINGS ${slicer_build_dir}/PACKAGES.txt package_list)
-
-          foreach(p ${package_list})
-            get_filename_component(package_name "${p}" NAME)
-            message(STATUS "Uploading URL to [${package_name}] on CDash")
-            slicer_ctest_upload_url(
-              ALGO "SHA512"
-              DOWNLOAD_URL_TEMPLATE "${SLICER_PACKAGE_MANAGER_URL}/api/v1/file/hashsum/%(algo)/%(hash)/download"
-              FILEPATH ${p}
-              )
-            if(run_ctest_submit)
-              ctest_submit(PARTS Upload)
-            endif()
-          endforeach()
+          if(run_ctest_with_upload)
```

Comment:
This extra check is not required as this code block is already conditionally executed based on:

```
        if(run_ctest_with_upload)
          message(STATUS "Uploading Slicer package URL ...")
```